crates/llms/src/chat/mistral.rs

```diff
@@ -300,32 +300,49 @@ impl MistralLlama {
         model_id: &str,
         arch: Option<&str>,
         hf_token_literal: Option<&Secret<String>>,
+        gguf_filename: Option<PathBuf>,
     ) -> Result<Self> {
         let model_parts: Vec<&str> = model_id.split(':').collect();
 
-        // Hardcoded model architecture can ensure correct loading type.
-        // If not provided, it will be inferred (generally from `.model_type` in a downloaded `config.json`)
-        let loader_type = arch
-            .map(|a| {
-                mistralrs::NormalLoaderType::from_str(a)
-                    .map_err(|e| ChatError::UnsupportedModelType { source: e.into() })
-            })
-            .transpose()?;
+        // Loading the GGUF directly (as if it is a quantized model, although it need not be quantized).
+        let loader: Result<Box<dyn Loader>> = if let Some(gguf) = gguf_filename {
+            Ok(GGUFLoaderBuilder::new(
+                None,
+                None,
+                model_parts[0].to_string(),
+                vec![gguf.to_string_lossy().to_string()],
+                GGUFSpecificConfig::default(),
+            )
+            .build())
+        } else {
+            // Hardcoded model architecture can ensure correct loading type.
+            // If not provided, it will be inferred (generally from `.model_type` in a downloaded `config.json`)
+            let loader_type = arch
+                .map(|a| {
+                    mistralrs::NormalLoaderType::from_str(a)
+                        .map_err(|e| ChatError::UnsupportedModelType { source: e.into() })
+                })
+                .transpose()?;
+
+            let builder = NormalLoaderBuilder::new(
+                mistralrs::NormalSpecificConfig::default(),
+                None,
+                None,
+                Some(model_parts[0].to_string()),
+            );
+
+            builder
+                .build(loader_type)
+                .map_err(|e| ChatError::FailedToLoadModel { source: e.into() })
+        };
 
-        let builder = NormalLoaderBuilder::new(
-            mistralrs::NormalSpecificConfig::default(),
-            None,
-            None,
-            Some(model_parts[0].to_string()),
-        );
         let device = Self::get_device();
         let token_source = hf_token_literal.map_or(TokenSource::CacheToken, |secret| {
+            tracing::info!("Explicitly given a Huggingface token. Using instead of any system defaults (if any).");
```

Comment:
```suggestion
            tracing::debug!("A HuggingFace token was directly specified. The specified token will be used instead of any system/environment defaults.");
```

nit: this could be a debug log. wording.